pkg/discovery/oci.go

```diff
@@ -29,11 +32,23 @@ type OCIDiscovery struct {
 }
 
 // NewOCIDiscovery returns a new Discovery using the specified OCI image.
-func NewOCIDiscovery(name, image string) Discovery {
+func NewOCIDiscovery(name, image string, criteria *PluginDiscoveryCriteria) Discovery {
 	if config.IsFeatureActivated(constants.FeatureCentralRepository) {
+		// The plugin inventory uses relative image URIs to be future-proof.
+		// Determine the image prefix from the main image.
+		// E.g., if the main image is at project.registry.vmware.com/tanzu-cli/plugins/plugin-inventory:latest
+		// then the image prefix should be project.registry.vmware.com/tanzu-cli/plugins/
+		imagePrefix := path.Dir(image)
+		// The data for the inventory is stored in the cache
+		pluginDataDir := filepath.Join(common.DefaultCacheDir, inventoryDirName, name)
```

Comment:
Notice this change where the inventory SQLite DB is moved from:
`~/.cache/_tanzu/plugin_inventory/plugin_inventory.db` to
`~/.cache/_tanzu/plugin_inventory/<discoveryName>/plugin_inventory.db`

This is to allow having multiple discovery sources using SQLite at the same time.

@anujc25 Does that fits with your publication approach?

Reply:
Yes. This seems correct.

Reply:
Yes. This seems reasonable.